extensions/amp-story/0.1/media-pool.js

```diff
@@ -205,21 +205,28 @@ export class MediaPool {
     this.forEachMediaType_(key => {
       const type = MediaType[key];
       const count = maxCounts[type] || 0;
+
+      // Small optimization. Cloning nodes is faster than building them, so
+      // constructing a seed element makes sense for initialization.
+      const seed = this.mediaFactory_[type].call(this);
+
       this.allocated[type] = [];
       this.unallocated[type] = [];
-      for (let i = 0; i < count; i++) {
-        this.vsync_.mutate(() => {
-          const mediaEl = this.mediaFactory_[type].call(this);
-          mediaEl.setAttribute('pool-element', elId++);
+
+      this.vsync_.mutate(() => {
+        for (let i = count; i > 0; i--) {
+          // Re-use seed when reaching end of set.
+          const mediaEl = (i == 1) ? seed : seed.cloneNode();
           const sources = this.getDefaultSource_(type);
+          mediaEl.setAttribute('pool-element', elId++);
           this.enqueueMediaElementTask_(mediaEl,
```

Comment:
I'm not sure if we can enqueue this task for the seed and have it also work for the cloned nodes and I didn't wanna perf-test it so I opened #14058